tensorflow/core/grappler/optimizers/data/noop_elimination_test.cc

```diff
@@ -51,6 +68,23 @@ NodeDef *MakeUnaryNonConstNode(StringPiece node_type, string input_node,
                               GetCommonAttributes(), graph);
 }
 
+NodeDef *MakeShardNonConstNode(string input_node, MutableGraphView *graph) {
```

Comment:
Instead of introducing `MakeShardNonConstNode`, generalize `MakeNonConstNode` to take a list of DTYPES to use to create placeholders.

Reply:
Updated.